prExplorer/client-app/src/features/products/dashboard/ProductList.tsx

```diff
@@ -1,38 +1,52 @@
 import React, { SyntheticEvent, useState } from "react";
-import { Product } from "../../../app/models/product";
 import { Button, Item, Label, Segment } from "semantic-ui-react";
+import { useStore } from "../../../app/stores/store";
+import { observer } from "mobx-react-lite";
 
-interface Props {
-  products: Product[];
-  selectProduct: (id: string) => void;
-  deleteProduct: (id: string) => void;
-  submitting: boolean;
-}
+export default observer(function ProductList() {
+  const { productStore } = useStore();
+  const { deleteProduct, productsSortedByName, loading } = productStore;
 
-export default function ProductList({ products, selectProduct, deleteProduct, submitting }: Props) {
+  const [target, setTarget] = useState("");
```

Comment:
For what do you need this `target` in state? Isn't `selectedProduct`  in your store enough?